### ISSUE TITLE
TypeError: Argument 'currency' has incorrect type (expected nautilus_trader.model.currency.Currency, got NoneType)

### ISSUE THREAD

**Issue by graceyangfan** [nautechsystems/nautilus_trader#597]
# Bug Report
when running binance_futures_market_maker,the living code seems can not get the account curreny from the exchange.
### Expected Behavior
Add here...
I  have some usdt in the contract account,I just change the API key and secrect to my  personal account.
### Actual Behavior
here is what I get when I running the code:
  
```   balances=parse_account_balances_http(assets=account_info.assets),
  File "/home/ubuntu/.local/lib/python3.8/site-packages/nautilus_trader/adapters/binance/futures/parsing/account.py", line 36, in parse_account_balances_http
    total=Money(total, currency),
TypeError: Argument 'currency' has incorrect type (expected nautilus_trader.model.currency.Currency, got NoneType)
2022-04-01T15:22:21.687495202Z [WRN] REAL-001.TradingNode: Timed out (5.0s) waiting for engines to connect and initialize.
Status
------
DataEngine.check_connected() == True
ExecEngine.check_connected() == False
Traceback (most recent call last):
  File "nautilus_trader/common/component.pyx", line 522, in nautilus_trader.common.component.Component.dispose
  File "nautilus_trader/common/component.pyx", line 616, in nautilus_trader.common.component.Component._trigger_fsm
  File "nautilus_trader/common/component.pyx", line 613, in nautilus_trader.common.component.Component._trigger_fsm
  File "nautilus_trader/core/fsm.pyx", line 115, in nautilus_trader.core.fsm.FiniteStateMachine.trigger
nautilus_trader.core.fsm.InvalidStateTrigger: RUNNING -> DISPOSE

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "binance_futures_market_maker.py", line 104, in <module>
    node.dispose()
  File "/home/ubuntu/.local/lib/python3.8/site-packages/nautilus_trader/live/node.py", line 414, in dispose
    self._data_engine.dispose()
  File "nautilus_trader/common/component.pyx", line 502, in nautilus_trader.common.component.Component.dispose
  File "nautilus_trader/common/component.pyx", line 531, in nautilus_trader.common.component.Component.dispose
  File "nautilus_trader/common/component.pyx", line 616, in nautilus_trader.common.component.Component._trigger_fsm
  File "nautilus_trader/common/component.pyx", line 613, in nautilus_trader.common.component.Component._trigger_fsm
  File "nautilus_trader/core/fsm.pyx", line 115, in nautilus_trader.core.fsm.FiniteStateMachine.trigger
nautilus_trader.core.fsm.InvalidStateTrigger: RUNNING -> DISPOSED
```

### Steps to Reproduce the Problem

1.input  the API key and secret
2.run  binance_futures_market_maker.py

### Specifications

- OS platform:Ubuntu 20.04
- Python version: 3.8 
- `nautilus_trader` version:1.140.0

**Comment by cjdsellers** [comment]
Hi @graceyangfan thanks for the report!

This has been fixed in version `1.141.0` currently on the `develop` branch. We'll be merging and releasing this version very soon (possibly today or at least this weekend).

**Comment by graceyangfan** [comment]
thanks for your work
